### PR TITLE
Rename CoreFx.Private.TestUtilities and reference it in slns

### LINKED ISSUE #31876: Consider renaming CoreFx.Private.TestUtilities
We are called .NET Libraries Team 🤙 

cc @danmosemsft @stephentoub

### PR DESCRIPTION
**Pull request by ViktorHofer**
This fixes the issue where VS refuses to build the test library because it has a reference to TestUtilities which isn't included in the sln. That wasn't a problem before but it seems now that we turned on project restore for everything or because we are using nuget static graph evaluation VS isn't happy about this anymore and errors.

Also I'm renaming CoreFx.Private.TestUtilities to TestUtilities as we don't want to use the `corefx` term anymore and don't need `private` in it anymore as well as we don't produce a package for that library anymore.

I didn't rename CoreFx.Private.TestUtilities.Unicode as I don't have any context about that test library but I hope @GrabYourPitchforks will do the same for it.

Fixes https://github.com/dotnet/runtime/issues/31876

### CODE REVIEW

**Comment by ViktorHofer** [comment]
cc @GrabYourPitchforks

**Comment by ViktorHofer** [comment]
That's the script that I ran:

```
foreach ($slnFile in Get-ChildItem -Recurse -Path "*" -Filter *.sln)
{
    $parentDir = (get-item $slnFile.FullName).Directory.FullName
    $testsDir = Join-Path $parentDir tests
    if (Test-Path $testsDir)
    {
        dotnet sln $slnFile.FullName add -s tests Common\tests\CoreFx.Private.TestUtilities\CoreFx.Private.TestUtilities.csproj
    }
}
```

File: src/mono/netcore/gen-xunit-runner/Program.cs [hunk 1]
```diff
@@ -284,7 +284,7 @@ static int Main (string[] args)
 
 		// Despite a lot of effort, couldn't get dotnet to load these assemblies from the sdk dir, so copy them to our binary dir
 //		File.Copy ($"{sdkdir}/Microsoft.DotNet.PlatformAbstractions.dll", AppContext.BaseDirectory, true);
-		File.Copy ($"{sdkdir}/CoreFx.Private.TestUtilities.dll", AppContext.BaseDirectory, true);
+		File.Copy ($"{sdkdir}/TestUtilities.dll", AppContext.BaseDirectory, true);
```
Comment (safern) [inline src/mono/netcore/gen-xunit-runner/Program.cs:287]: Curious, what is this program for?

File: src/libraries/System.Net.Security/System.Net.Security.sln [hunk 2]
```diff
@@ -1,4 +1,4 @@
-Microsoft Visual Studio Solution File, Format Version 12.00
+﻿Microsoft Visual Studio Solution File, Format Version 12.00
```
Comment (safern) [inline src/libraries/System.Net.Security/System.Net.Security.sln:1]: Would changing the BOM marker change anything?

**Review by safern — APPROVED** [review]
LGTM. Question, should we also rename: `CoreFx.Private.TestUtilities.Unicode`, to `TestUtilities.Unicode` ?

File: src/mono/netcore/gen-xunit-runner/Program.cs (hunk 1 above)
Comment (ViktorHofer) [reply src/mono/netcore/gen-xunit-runner/Program.cs:287]: @stephentoub

**Comment by ViktorHofer** [comment]
> I didn't rename CoreFx.Private.TestUtilities.Unicode as I don't have any context about that test library but I hope @GrabYourPitchforks will do the same for it.

File: src/libraries/System.Net.Security/System.Net.Security.sln (hunk 2 above)
Comment (ViktorHofer) [reply src/libraries/System.Net.Security/System.Net.Security.sln:1]: Unsure, but I don't care much assuming we won't change these files again anytime soon (now that there is no target framework in them anymore).

**Comment by GrabYourPitchforks** [comment]
Feel free to rename the `Unicode` project as well per Santi's suggestion. Otherwise if you don't get to it as part of this PR I'll send a separate PR for it.

**Comment by ViktorHofer** [comment]
Failures are known infra ones.

**Comment by ViktorHofer** [comment]
@GrabYourPitchforks please send one, I'll be OOF rest of the week.

File: src/mono/netcore/gen-xunit-runner/Program.cs (hunk 1 above)
Comment (akoeplinger) [reply src/mono/netcore/gen-xunit-runner/Program.cs:287]: It's a tool that converts xunit tests into a simple app that doesn't depend on the xunit runner, we use it for bringing up webassembly on netcore mono as a lot of the xunit machinery doesn't work there.

Comment (safern) [reply src/mono/netcore/gen-xunit-runner/Program.cs:287]: I see. Didn't know if this was for pre-consolidation.

**Pull request merged**